### PR TITLE
Add initial ztunnel cross-version test

### LINKED ISSUE #52701: Ambient cross version compatibility
From [Ambient GA RFC](https://docs.google.com/document/d/1GiD2-MmY9VrkfEs-ET0b7UGPmVII2GQUD7m1iAvbrjo):
> For proxy skew with Ztunnel and waypoints. This is likely already done, just need to make sure our documentation and testing is high enough quality.

### PR DESCRIPTION
**Pull request by therealmitchconnors**
**Please provide a description of this PR:**
Fixes #52701 Adding cross-version compat tests for ztunnel with CNI and Istiod

### CODE REVIEW

File: tests/integration/helm/upgrade/util.go [hunk 1]
```diff
@@ -168,6 +168,57 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
...
+		h := helm.New(cs.Filename())
+		nsConfig := helmtest.DefaultNamespaceConfig
+		t.CleanupConditionally(func() {
+			// only need to do call this once as helm doesn't need to remove
+			// all versions
+			helmtest.DeleteIstio(t, h, cs, nsConfig, isAmbient)
+		})
+		s := t.Settings()
+		prevVariant := s.Image.Variant
+		// Istio 1.21 ambient did not support distroless, always use debug.
```
Comment (bleggett) [inline tests/integration/helm/upgrade/util.go:186]: Yeah I think we can drop this now, it's no longer true.

File: tests/util/sanitycheck/sanity_check.go [hunk 2]
```diff
@@ -83,6 +88,36 @@ func setupTrafficTest(t framework.TestContext, revision string, ambient bool) (n
 	return testNs, client, server
 }
 
+func BlockTestWithPolicy(t framework.TestContext, client, server echo.Instance) {
+	ns := server.Config().Namespace.Name()
+	_, err := server.Config().Cluster.Istio().SecurityV1().AuthorizationPolicies(ns).Create(
```
Comment (howardjohn) [inline tests/util/sanitycheck/sanity_check.go:93]: nit: normally we would do something like `t.ConfigIstio().Eval(...).Apply()`. Not a huge deal though

Comment (howardjohn) [reply tests/util/sanitycheck/sanity_check.go:93]: Although this won't cleanup which may break things later? So maybe it is a big deal

**Review by howardjohn — COMMENTED** [review]
Nice!

File: tests/integration/helm/upgrade/util.go [hunk 3]
```diff
@@ -168,6 +168,69 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
...
+		// TODO(https://github.com/istio/istio/issues/50387) remove this, always use s.Image.Variant
+		if isAmbient {
+			prevVariant = "debug"
+		}
```
Comment (bleggett) [inline tests/integration/helm/upgrade/util.go:190]:
```suggestion
```

I think this entire block can/should be removed? For at least the last 2 releases the `debug` override should definitely not be required.

File: tests/integration/helm/util.go [hunk 4]
```diff
@@ -274,11 +274,6 @@ func InstallIstio(t framework.TestContext, cs cluster.Cluster, h *helm.Helm, ove
...
 		// Prepend ~ to the version, so that we can refer to the latest patch version of a minor version
 		versionArgs = fmt.Sprintf("--repo %s --version ~%s", t.Settings().HelmRepo, version)
-		// Currently the ambient in-place upgrade tests try an upgrade from previous release which is 1.20,
-		// and many of the profile override values seem to be unrecognized by the gateway installation.
-		// So, this is a workaround until we move to 1.21 where we can use --set profile=ambient for the install/upgrade.
-		// TODO: Remove this once the previous release version for the test becomes 1.21
-		// refer: https://github.com/istio/istio/issues/49242
 		if ambientProfile {
 			gatewayOverrideValuesFile = GetValuesOverrides(t, t.Settings().Image.Hub, version, t.Settings().Image.Variant, "", false)
 		}
```
Comment (bleggett) [inline tests/integration/helm/util.go:279]:
```suggestion
```

If the comment no longer applies, then the override should also be removed, right?

File: tests/integration/helm/upgrade/util.go [hunk 5]
```diff
@@ -168,6 +168,64 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
...
+		})
+		s := t.Settings()
+		prevVariant := s.Image.Variant
+		overrideValuesFile := helmtest.GetValuesOverrides(t, gcrHub, "", prevVariant, "", isAmbient)
+		// todo tag version is not helm version
+		helmtest.InstallIstio(t, cs, h, overrideValuesFile, previousVersion, true, isAmbient, nsConfig)
+		helmtest.VerifyInstallation(t, cs, nsConfig, true, isAmbient, "")
+
+		_, oldClient, oldServer := sanitycheck.SetupTrafficTestAmbient(t, "")
+		sanitycheck.RunTrafficTestClientServer(t, oldClient, oldServer)
```
Comment (bleggett) [inline tests/integration/helm/upgrade/util.go:192]:
```suggestion
		correctnessCheck.RunTrafficTestClientServer(t, oldClient, oldServer)
```

meganit: Can we avoid the use of phrases like `sanity check` in the codebase/examples - `check/verify/correctness-check` etc are probably better.

(I'm rather bad at remembering to do this myself, which is the only reason I'm mentioning it here).

**Review by bleggett — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**